pylxd/tests/test_client.py

```diff
@@ -72,6 +73,17 @@ def test_create_endpoint(self):
 
         self.assertEqual(expected, an_client.api._api_endpoint)
 
+    def test_create_endpoint_unixsocket(self):
+        """Test with unix socket endpoint."""
+        endpoint = '/tmp/unix.socket'
+        expected = 'http+unix://%2Ftmp%2Funix.socket/1.0'
+
+        if not os.path.isfile(endpoint):
+            open(endpoint, 'a').close()
```

Comment:
You can mock out `os.path.isfile` to return True in this case. That'd be preferred. We don't want to have to touch the filesystem in tests at all if we can help it.